.github/workflows/rust.yml

```diff
@@ -3,7 +3,12 @@
 #
 name: Rust
 
-on: [ push, pull_request ]
+on:
+  push:
+    branches:
```

Comment:
So if I understand correctly, the reason this is required is because GitHub ridiculously includes tags in the push event, and we had no filter at all here to begin with, but now we explicitly restrict it to this set of branches?

Do we want to, perhaps, do it for all branches (just not tags)?  That's how buildomat works for example.

Reply:
I'm fine either way; I think the vast majority of push events to other branches are duplicated with the pull request events, and we need the pull request events to handle PRs from forks.

Another approach might be to run the workflow on any push to a branch, and only run pull request workflows if they come from a fork. I don't know if that's possible.

Reply:
In practice I think we would only particularly care if an artefact we expect to download and run (for a non-PR work in progress branch) was missing.  Those come from buildomat, not GHA, so I expect making this change is likely fine as-is!

Reply:
(have you had a chance to run the numbers on cost/time savings if buildomat did filter out non-PR branches, btw?)